openmls/src/credentials/mod.rs

```diff
@@ -412,11 +367,42 @@ mod unit_tests {
             deserialized.serialized_content()
         );
 
-        let deserialized_basic_credential = BasicCredential::try_from(&deserialized).unwrap();
+        let deserialized_basic_credential = BasicCredential::try_from(deserialized).unwrap();
         assert_eq!(
             deserialized_basic_credential.identity(),
             IDENTITY.as_bytes()
         );
         assert_eq!(basic_credential, deserialized_basic_credential);
     }
+
+    /// Test the [`Credential`] with a custom credential.
+    #[test]
+    fn custom_credential() {
+        #[derive(
+            Debug, Clone, PartialEq, Eq, TlsSize, TlsSerialize, TlsDeserialize, TlsDeserializeBytes,
+        )]
+        struct CustomCredential {
+            inner: VLBytes,
```

Comment:
Can you add another field here to make sure it's working for other things that are not just bytes?